Formula/xpipe.rb

```diff
@@ -0,0 +1,17 @@
+class Xpipe < Formula
+  desc "Split input and feed it into the given utility"
+  homepage "https://www.netmeister.org/apps/xpipe.html"
+  url "http://www.netmeister.org/apps/xpipe-1.0.tar.gz"
+  sha256 "6f15286f81720c23f1714d6f4999d388d29f67b6ac6cef427a43563322fb6dc1"
+  license "BSD-2-Clause"
+
+  def install
+    system "make"
+    bin.install "xpipe"
+    man1.install "doc/xpipe.1"
+  end
+
+  test do
+    system "true"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The tool has a battery of tests, but they require creation of sample data.  All that can be run via 'make test', but it seems at 'test' time here we no longer are in the extracted tarball directory?  Is there an easy way to say "cd extracted-dir && make test" here?

Reply:
FWIW, I added a basic check for core functionality. Not as exhaustive as the full test suite, but should be sufficient to verify the tool as installed actually works as intended.